drivers/i2c/i2c_ll_stm32.h

```diff
@@ -24,6 +24,7 @@ struct i2c_stm32_data {
 #ifdef CONFIG_I2C_STM32_INTERRUPT
 	struct k_sem device_sync_sem;
 #endif
+	struct k_sem		sem_lock;
```

Comment:
A comment would be good to explain functional diff between device_sync_sem and sem_lock

Reply:
device_sync_sem is not a mutex (i.e. to guarantee mutual exclusion). Instead, it is a semaphore whose purpose is to wait until data are ready. If you see the various i2c drivers (for example nios2)
all drivers uses both kinds of semaphores.

Reply:
Right, thanks.